sql/updates/post-update.sql

```diff
@@ -1,67 +1,107 @@
+    IF ts_major::INTEGER = 2 AND ts_minor::INTEGER >= 7 THEN
             CREATE PROCEDURE _timescaledb_internal.post_update_cagg_try_repair(
                 cagg_view REGCLASS, force_rebuild boolean
             ) AS '@MODULE_PATHNAME@', 'ts_cagg_try_repair' LANGUAGE C;
     END IF;
-    FOR vname, materialized_only IN select format('%I.%I', cagg.user_view_schema, cagg.user_view_name)::regclass, cagg.materialized_only from _timescaledb_catalog.continuous_agg cagg
-    LOOP
-        -- the cast from oid to text returns
-        -- quote_qualified_identifier (see regclassout).
-        --
-        -- We use the if statement to handle pre-2.0 as well as
-        -- post-2.0.  This could be turned into a procedure if we want
-        -- to have something more generic, but right now it is just
-        -- this case.
-        IF ts_version < '2.0.0' THEN
-            EXECUTE format('ALTER VIEW %s SET (timescaledb.materialized_only=%L) ', vname::text, materialized_only);
-        ELSIF ts_version < '2.7.0' THEN
-            EXECUTE format('ALTER MATERIALIZED VIEW %s SET (timescaledb.materialized_only=%L) ', vname::text, materialized_only);
-        ELSE
-            SET log_error_verbosity TO VERBOSE;
-            CALL _timescaledb_internal.post_update_cagg_try_repair(vname, false);
-        END IF;
-    END LOOP;
-    IF ts_version >= '2.7.0' THEN
+    IF ts_major::INTEGER = 2 AND ts_minor::INTEGER < 7 THEN
+        FOR vname, materialized_only IN select format('%I.%I', cagg.user_view_schema, cagg.user_view_name)::regclass, cagg.materialized_only from _timescaledb_catalog.continuous_agg cagg
+        LOOP
+            -- the cast from oid to text returns
+            -- quote_qualified_identifier (see regclassout).
+            --
+            -- We use the if statement to handle pre-2.0 as well as
+            -- post-2.0.  This could be turned into a procedure if we want
+            -- to have something more generic, but right now it is just
+            -- this case.
+            IF ts_minor::INTEGER < 0 THEN
+                EXECUTE format('ALTER VIEW %s SET (timescaledb.materialized_only=%L) ', vname::text, materialized_only);
+            ELSIF ts_minor::INTEGER < 7 THEN
+                EXECUTE format('ALTER MATERIALIZED VIEW %s SET (timescaledb.materialized_only=%L) ', vname::text, materialized_only);
+            END IF;
+        END LOOP;
+    ELSIF ts_major::INTEGER = 2 AND ts_minor::INTEGER >= 7 THEN
+        FOR vname, mat_ht_id, materialized_only, finalized IN select format('%I.%I', cagg.user_view_schema,
+            cagg.user_view_name)::regclass, cagg.mat_hypertable_id, cagg.materialized_only, cagg.finalized from _timescaledb_catalog.continuous_agg cagg
+        LOOP
+            IF ts_minor::INTEGER < 12 THEN
+                SET log_error_verbosity TO VERBOSE;
+                CALL _timescaledb_internal.post_update_cagg_try_repair(vname, false);
+            ELSIF ts_minor::INTEGER = 12 AND finalized IS FALSE THEN
```

Comment:
I don't understand the reason for the following lines at the moment. Could you add a comment on why we call  `public.cagg_migrate` for this specific version?

Reply:
Fixed!